src/Microsoft.Dnx.Tooling/Publish/PublishProject.cs

```diff
@@ -375,54 +361,85 @@ private bool PrunePackages(PublishRoot root)
                 foreach (var library in target.Libraries)
                 {
                     var packagesDir = resolver.GetInstallPath(library.Name, library.Version);
-                    var manifest = resolver.GetManifestFilePath(library.Name, library.Version);
-
-                    keep.Add(manifest);
 
                     foreach (var path in library.RuntimeAssemblies)
                     {
-                        keep.Add(CombinePath(packagesDir, path));
+                        keep.Add(Path.Combine(packagesDir, Path.GetDirectoryName(path)));
                     }
 
                     foreach (var path in library.CompileTimeAssemblies)
                     {
-                        keep.Add(CombinePath(packagesDir, path));
+                        keep.Add(Path.Combine(packagesDir, Path.GetDirectoryName(path)));
                     }
 
                     foreach (var path in library.NativeLibraries)
                     {
-                        keep.Add(CombinePath(packagesDir, path));
+                        keep.Add(Path.Combine(packagesDir, Path.GetDirectoryName(path)));
                     }
 
                     foreach (var path in library.ResourceAssemblies)
                     {
-                        keep.Add(CombinePath(packagesDir, path));
+                        keep.Add(Path.Combine(packagesDir, Path.GetDirectoryName(path)));
                     }
+                }
+            }
 
-                    foreach (var specialFolder in specialFolders)
-                    {
-                        var specialFolderPath = CombinePath(packagesDir, specialFolder);
+            var frameworkSpecificContents = new List<string>
+            {
+                "lib",
+                "ref"
+            };
 
-                        if (!Directory.Exists(specialFolderPath))
-                        {
-                            continue;
-                        }
+            var extensionsToPrune = new List<string>
+            {
+                "xml"
+            };
 
-                        keep.AddRange(Directory.EnumerateFiles(specialFolderPath, "*.*", SearchOption.AllDirectories));
-                    }
-                }
-            }
+            var exemptFrameworks = new List<FrameworkName>
+            {
+                VersionUtility.ParseFrameworkName("native")
```

Comment:
Currently the only recognized but potentially indirectly referenced `TFM`. Note that currently `xamarinmac20` and `xamarinios10` are common unrecognized `TMF`s and will not be pruned.